addons/binding/org.openhab.binding.networkhealth/src/main/java/org/openhab/binding/networkhealth/discovery/NetworkHealthDiscoveryService.java

```diff
@@ -0,0 +1,236 @@
+
+		try {
+			// For each interface ...
+			for (Enumeration<NetworkInterface> en = NetworkInterface
+					.getNetworkInterfaces(); en.hasMoreElements();) {
+				NetworkInterface networkInterface = en.nextElement();
+				if (!networkInterface.isLoopback()) {
+
+					// .. and for each address ...
+					for (Iterator<InterfaceAddress> it = networkInterface
+							.getInterfaceAddresses().iterator(); it.hasNext();) {
+						
+						// ... get IP and Subnet 
+						InterfaceAddress interfaceAddress = it.next();
+						interfaceIPs.add(interfaceAddress.getAddress()
+								.getHostAddress()
+								+ "/"
+								+ interfaceAddress.getNetworkPrefixLength());
+					}
+				}
+			}
+		} catch (SocketException e) {
+		}
+
+		return interfaceIPs;
+	}
+	
+	/**
+	 * Takes the interfaceIPs and fetches every IP which can be assigned on their network
+	 * @param networkIPs The IPs which are assigned to the Network Interfaces
+	 * @return Every single IP which can be assigned on the Networks the computer is connected to
+	 */
+	private Queue<String> getNetworkIPs(TreeSet<String> interfaceIPs) {
+		Queue<String> networkIPs = new LinkedBlockingQueue<String>();
+
+		for (Iterator<String> it = interfaceIPs.iterator(); it.hasNext();) {
+			try {
+				// gets every ip which can be assigned on the given network
+				SubnetUtils utils = new SubnetUtils(it.next());
+				String[] addresses = utils.getInfo().getAllAddresses();
+				for (int i = 0; i < addresses.length; i++) {
+					networkIPs.add(addresses[i]);
+				}
+
+			} catch (Exception ex) {
+			}
+		}
+
+		return networkIPs;
+	}
+
+	public Set<ThingTypeUID> getSupportedThingTypes() {
+		return SUPPORTED_THING_TYPES_UIDS;
+	}
+
+	@Override
+	protected void startBackgroundDiscovery() {
+		new Thread(new Runnable() {
+			@Override
+			public void run() {
+				discoverNetwork();
+			}
+		}).start();
+	}
+	
+	/**
+	 * Starts the DiscoveryThread for each IP on the Networks
+	 * @param allNetworkIPs
+	 */
+	private void startDisconvery(final Queue<String> networkIPs) {
+		
+		Runnable runnable = new Runnable() {
+			public void run() {
+				DiscoveryThread discoveryThread = null;
+				DiscoveryThreadResult discoveryThreadResult = new DiscoveryThreadResult() {
+					@Override
+					public void newDevice(String ip) {
+						submitDiscoveryResults(ip);
+					}
+				};
+				
+				// ensures that only one thread at  a time access the queue
+				synchronized (lockObject) {
+					if( networkIPs.isEmpty()) {
+						discoveryJob.cancel(false);
+					} else {
+						discoveryThread = new DiscoveryThread(networkIPs.remove(), discoveryThreadResult);						
+					}
+				}
+				
+				if( discoveryThread != null )
+					discoveryThread.start();
+			}
+		};
+		
+		/* Every milisecond a new thread will be created. Due to the fact that the PING has a timeout of 1 sec,
+		 * only about 1000 Threads will be create at max */
+		discoveryJob = scheduler.scheduleAtFixedRate(runnable, 0, 1, TimeUnit.MILLISECONDS);

```

Comment:
Unfortunately I don't know how to make it configurable. For me their seems to be no way of asking the user to specify further Scan Parameters. 
I haven't tested it on a PI either. The Problem is that a network which has a subnet mask of 16 has 65025 IP which could be assigned to Devices and they could be even larger. Increasing the Thread creating Time means a longer Scan. I'm not sure if the user wants to wait about 15 min for his 3 Devices to appear (Currently it is set to 5min). 
Maybe it would make sense to be able to dynamically assign the timeout for the scan, something like a function which returns the approximate Scan Time for this run. The Timeout and the creating time caught depend on the current network size and cpu power.